### PR TITLE
refactor(custmsg+wasm): adjust message emitter iface

### PR DESCRIPTION
**Pull request by MStreet3**
passing the `custmsg.Labeler` to `host.ModuleConfig.Labeler` is blocked because the types do not match.  this PR fixes the type mismatch:

* removes the `MessageEmitter` interface from `wasm/host` as it is self-referential and therefore cannot be satisfied by an implementation that does not import `wasm/host`.

* `MessageEmitter` is instead defined in `custmsg` and imported into `wasm/host`.  Now `custmsg.Labeler` implements `custmsg.MessageEmitter` and can be passed directly to `host.ModuleConfig`.

### CODE REVIEW

**Review by cedric-cordenier — APPROVED** [review]

File: pkg/custmsg/custom_message.go [hunk 1]
```diff
@@ -11,17 +11,31 @@ import (
...
-func NewLabeler() Labeler {
-	return Labeler{labels: make(map[string]string)}
+func NewLabeler() labeler {
+	return labeler{labels: make(map[string]string)}
 }
 
 // WithMapLabels adds multiple key-value pairs to the CustomMessageLabeler for transmission
 // With SendLogAsCustomMessage
-func (l Labeler) WithMapLabels(labels map[string]string) Labeler {
+func (l labeler) WithMapLabels(labels map[string]string) MessageEmitter {
```
Comment (patrickhuie19) [inline pkg/custmsg/custom_message.go:38]: idiomatic go to accept interfaces as params and return structs. could we do that here?

File: pkg/custmsg/custom_message.go [hunk 2]
```diff
@@ -38,7 +52,7 @@ func (l Labeler) WithMapLabels(labels map[string]string) Labeler {
 }
 
 // With adds multiple key-value pairs to the CustomMessageLabeler for transmission With SendLogAsCustomMessage
-func (l Labeler) With(keyValues ...string) Labeler {
+func (l labeler) With(keyValues ...string) MessageEmitter {
```
Comment (patrickhuie19) [inline pkg/custmsg/custom_message.go:55]: same comment as above

File: pkg/custmsg/custom_message.go (hunk 1 above)
Comment (MStreet3) [reply pkg/custmsg/custom_message.go:38]: not a hard axiom.  In this case there is an exception for two reasons:
1. `labeler` is the only implementation of `MessageEmitter` (see Effective Go on [generality](https://go.dev/doc/effective_go#generality) of the interface rule)
2. the interface is self-referential, if the methods don't return `MessageEmitter` then `labeler` won't implement `MessageEmitter`

Comment (MStreet3) [reply pkg/custmsg/custom_message.go:38]: `Labeler` was originally public, but all of its fields are private, which makes it redundant to the interface so I made it private.  That's a third reason to not return `labeler` directly.

**Review by justinkaseman — DISMISSED** [review]

File: pkg/custmsg/custom_message.go (hunk 1 above)
Comment (patrickhuie19) [reply pkg/custmsg/custom_message.go:38]: Let's keep `Labeler` public for continued direct consumption by the engine and registry syncer in core.

**Review by patrickhuie19 — APPROVED** [review]

**Review by justinkaseman — APPROVED** [review]

**Pull request merged**